src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -816,78 +835,39 @@ public function setSession(SessionInterface $session)
     }
 
     /**
-     * Returns the client IP addresses.
-     *
-     * In the returned array the most trusted IP address is first, and the
-     * least trusted one last. The "real" client IP address is the last one,
-     * but this is also the least trusted one. Trusted proxies are stripped.
-     *
-     * Use this method carefully; you should use getClientIp() instead.
-     *
-     * @return array The client IP addresses
+     * @see IpRetriever\IpRetriever\IpRetrieverInterface::getClientIps()
      *
      * @see getClientIp()
+     *
+     * @deprecated since version 2.7, to be removed in 3.0.
```

Comment:
Every time you deprecate a method, you must call `trigger_error` like the following: https://github.com/symfony/symfony/blob/d94d837e9ea75d76eeb0a43e0535a5e7a7a01542/src/Symfony/Component/HttpKernel/Tests/Logger.php#L94